File: leet_SLWD_1456_max_vowels.ts
function maxVowels(s: string, k: number): number {
    let v = 'aeiou'
    let L = 0, R = -1
    let temp = 0
    let res = 0
    while (++R < s.length) {
        if (v.includes(s[R])) {
            ++temp
        }
        if (R - L == k) {
            if (v.includes(s[L])) {
                --temp
            }
            ++L
        }
        res = res > temp ? res : temp
    }
    return res

};

File: leet_SLWD_1456_max_vowels.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        v = 'aeiou'
        res = 0
        temp = 0
        l = 0
        for r, c in enumerate(s):
            if c in v:
                temp += 1
            if r - l == k:
                if s[l] in v:
                    temp -= 1
                l += 1
            res = max(res, temp)
        return res
